config/loader.go

```diff
@@ -12,7 +19,38 @@ type Config struct {
 	}
 }
 
-func LoadString(tomlText string) (*Config, error) {
+type ConfigLoader struct {
+	env map[string]string
+}
+
+func NewConfigLoader() ConfigLoader {
+	envrionment := map[string]string{}
+	for _, e := range os.Environ() {
+		pair := strings.Split(e, "=")
+		envrionment[pair[0]] = pair[1]
+	}
```

Comment:
Can we extract this logic to a standalone function, maybe even in `utils/`? You call it and get an array of environment variables as `map[string]string`.